packages/client-search/src/types/MultipleQueriesQuery.ts

```diff
@@ -16,3 +23,17 @@ export type MultipleQueriesQuery = {
    */
   readonly query?: string;
 };
+
+export type MultipleQueriesQuery = SharedMultipleQueriesQuery &
+  (
+    | {
+        readonly type?: 'default';
```

Comment:
this being optional means that `facet` is allowed without type. I'm not sure how to solve this, it seems like a limitation of TS

https://www.typescriptlang.org/play?#code/C4TwDgpgBAygFgQwE4QCZQLxQN5QJYB2qEAHgHIIC2EAXFAM7BKEDmUAvgNwBQ3okUAPIEAxtCzxkaKADJuUBVAAU8xWoA+OKPwgB+OgQgA3CEg6q1CzdguW1OugHJiAMwQBXADbBHPOwvZbDRwg-wcoRzcxHz9-NQB6ACpE0P9EqAAVOGgoiGAoAioIADpUu0T4srVc4DpGZgIWWLt2AEoebnj4qAB3OBAoVAB7AqHDKCGXbWz6aEQTKFMkIaR6XW4RMcYJ0VohXcwtQmJyIqdgCEZHDh4gA

https://stackoverflow.com/questions/41495369/typescript-discriminated-unions-with-optional-values

Reply:
https://www.typescriptlang.org/play?#code/C4TwDgpgBAygFgQwE4QCZQLxQN5QJYB2qEAHgHIIC2EAXFAM7BKEDmUAvgNwBQ3okUAPIEAxtCzxkaKADJuUBVAAU8xWoA+OKP1pQCEAG4QkHVWoWbsZ82p10A5MQBmCAK4AbYPZ42F7axo4Ab52UPYuYl4+vmoA9ABU8cG+8VAAKnDQERDAelQQAHTJNvGxxWrZwHSMzAQs0TbsAJQ83LGxUADucCBQqAD2ev36UP1O2pn00IhGUMZI-Uj0APzcIsOMo6K6wmKYWoTE5PkOwBCM9gA0UJUOCPYcPEA

it sort of works?

Reply:
the problem is that it allows `const once: Once = { indexName: 'test', facet: "lol" };` which is invalid

Reply:
From my link, it gives an error message with that. 

https://www.typescriptlang.org/play?#code/C4TwDgpgBAygFgQwE4QCZQLxQN5QJYB2qEAHgHIIC2EAXFAM7BKEDmUAvgNwBQ3okUAPIEAxtCzxkaKADJuUBVAAU8xWoA+OKP1pQCEAG4QkHVWoWbsZ82p10A5MQBmCAK4AbYPZ42F7axo4Ab52UPYuYl4+vmoA9ABU8cG+8VAAKnDQERDAelQQAHTJNvGxxWrZwHSMzAQs0TbsAJQ83LGxUADucCBQqAD2ev36UP1O2pn00IhGUMZI-Uj0APzcIsOMo6K6wmKYWoTE5PkOwBCM9gA02uC6jhAuHl4cPOsEm8CdeGJ0u+IHRFIFGop3OXmulQc7n67nsL24QA

Reply:
Should it be like that? 

https://www.typescriptlang.org/play?#code/C4TwDgpgBAygFgQwE4QCZQLxQN5QJYB2qEAHgHIIC2EAXFAM7BKEDmUAvgNwBQ3okUMhABuEJJhz8IAfjoERY9rylQAonmBwxEgBTYpdAOTEAZggCuAG2CH2AH0nhaUQ2YDGEGwBoo7zzUZmAhZ2AEplJygAeQIPCXhkNCgAMigdIVFxB3VNMXDuNwB7AkYoYo8ARjoYuKxcQmJyKmdDYAhGQw5OKAB6HrKAawLi0vKIACZq2Og6-CJSCmojNo6fAxc-b18ED2Bl9psu3v7CoaKS4DLpgGYp2pw5xsWWla314wgzK0OuY8Hhi5XDwAFjuMweDQWzX2qyg702nV+fSgBEKANG0wArGCJPV5k0li5XoYfJsYT9usjUejLmNsdFprjHlDCa0DiS4U4jIUAO54ABWn0KeB5gpMiMp-WpQA

Reply:
yes, but that type doesn't allow `type` to be omitted: 

<img width="825" alt="Screenshot 2021-04-09 at 13 41 47" src="https://user-images.githubusercontent.com/6270048/114174761-5e58de00-9939-11eb-86b8-4a2df10d966c.png">


In the end the one in this PR might be the best

Reply:
Ah I misunderstood your intention.